packages/dm-core-plugins/src/header/components/UserInfoDialog.tsx

```diff
@@ -82,33 +89,43 @@ export const UserInfoDialog = (props: UserInfoDialogProps) => {
         )}
       </Dialog.CustomContent>
       <Dialog.Actions>
-        <Button
-          onClick={() => {
-            navigator.clipboard.writeText(token)
-            toast.success('Copied token to clipboard')
-          }}
-        >
-          Copy token to clipboard
-        </Button>
-        <Button
-          onClick={() =>
-            dmssAPI
-              .tokenCreate()
-              .then((response: AxiosResponse<string>) =>
-                setAPIKey(response.data)
-              )
-              .catch((error: any) => {
-                console.error(error)
-                toast.error('Failed to create personal access token')
-              })
-          }
-        >
-          Create API-Key
-        </Button>
-        <Button onClick={() => logOut()}>Log out</Button>
-        <Button variant="outlined" onClick={() => setIsOpen(false)}>
-          Cancel
-        </Button>
+        <FlexRow style={{ justifyContent: 'space-between', width: '100%' }}>
+          <FlexRow>
+            <Button
+              variant="ghost"
+              onClick={() => {
+                navigator.clipboard.writeText(token)
+                toast.success('Copied token to clipboard')
+              }}
+            >
+              Copy token to clipboard
+            </Button>
+            <Button
+              variant="ghost"
+              onClick={() =>
+                dmssAPI
+                  .tokenCreate()
+                  .then((response: AxiosResponse<string>) =>
+                    setAPIKey(response.data)
+                  )
+                  .catch((error: any) => {
+                    console.error(error)
+                    toast.error('Failed to create personal access token')
+                  })
+              }
+            >
+              Create API-Key
+            </Button>
+          </FlexRow>
+          <FlexRow>
+            <Button variant="ghost" color="danger" onClick={() => logOut()}>
+              Log out
+            </Button>
+            <Button onClick={() => setIsOpen(false)}>
+              {selectedRole !== originalRole.current ? 'Save' : 'Cancel'}
```

Comment:
See pitfalls of useRef: https://react.dev/reference/react/useRef
 "Don't read a ref during rendering"
 Also, this is still a bit misleading UI. Perhaps move the "setSelectedRole" to clicking the "Save" button. Then this ref wont be needed.

Reply:
right, my brain is still coupled to a different way of thinking about `ref`